cassandra/src/main/scala/akka/stream/alpakka/cassandra/scaladsl/CassandraFlow.scala

```diff
@@ -5,76 +5,122 @@
-   * share the same partition key. Cassandra unlogged batches that share the same partition key will only
-   * resolve to one write internally in Cassandra, boosting write performance.
+   * A flow writing to Cassandra for every stream element.
+   * The element is emitted unchanged.
    *
-   * Be aware that this stage does not preserve the upstream order.
+   * @param writeSettings settings to configure the write operation
+   * @param cqlStatement raw CQL statement
+   * @param statementBinder function to bind data from the stream element to the prepared statement
+   * @param session implicit Cassandra session from `CassandraSessionRegistry`
+   * @tparam T stream element type
    */
-  def createUnloggedBatchWithPassThrough[T, K](
-      parallelism: Int,
-      statement: PreparedStatement,
-      statementBinder: (T, PreparedStatement) => BoundStatement,
-      partitionKey: T => K,
-      settings: CassandraBatchSettings = CassandraBatchSettings()
-  )(implicit session: Session): Flow[T, T, NotUsed] = {
-    val graph = GraphDSL.create() { implicit builder =>
-      import GraphDSL.Implicits._
-
-      val groupStage: FlowShape[T, Seq[T]] =
-        builder.add(Flow[T].groupedWithin(settings.maxGroupSize, settings.maxGroupWait))
-
-      val groupByKeyStage: FlowShape[Seq[T], Seq[T]] =
-        builder.add(Flow[Seq[T]].map(_.groupBy(partitionKey).values.toList).mapConcat(identity))
-
-      val batchStatementStage: FlowShape[Seq[T], Seq[T]] = builder.add(
-        Flow[Seq[T]].mapAsyncUnordered(parallelism)(
-          list => {
-            val boundStatements = list.map(t => statementBinder(t, statement))
-            val batchStatement = new BatchStatement(BatchStatement.Type.UNLOGGED).addAll(boundStatements.asJava)
+  def create[T](
+      writeSettings: CassandraWriteSettings,
+      cqlStatement: String,
+      statementBinder: (T, PreparedStatement) => BoundStatement
+  )(implicit session: CassandraSession): Flow[T, T, NotUsed] = {
+    Flow
+      .lazyInitAsync { () =>
+        val prepare = session.prepare(cqlStatement)
+        prepare.map { preparedStatement =>
+          Flow[T].mapAsync(writeSettings.parallelism) { element =>
             session
-              .executeAsync(batchStatement)
-              .asScala()
-              .map(_ => list)(ExecutionContexts.sameThreadExecutionContext)
+              .executeWrite(statementBinder(element, preparedStatement))
+              .map(_ => element)(ExecutionContexts.sameThreadExecutionContext)
           }
-        )
-      )
-
-      val flattenResults: FlowShape[Seq[T], T] = builder.add(Flow[Seq[T]].mapConcat(_.toList))
-
-      groupStage ~> groupByKeyStage ~> batchStatementStage ~> flattenResults
+        }(session.ec)
+      }
+      .mapMaterializedValue(_ => NotUsed)
+  }
 
-      FlowShape(groupStage.in, flattenResults.out)
+  /**
+   * A flow writing to Cassandra for every stream element, passing context along.
+   * The element and context are emitted unchanged.
```

Comment:
> It might be worth mentioning that the context in this case is the element being persisted.

I'm not sure what you mean. The context can be anything.

> Is there any reason why we can't always use it instead of providing non-context factory method too?

That would require to have tuples as elements.

Reply:
> I'm not sure what you mean. The context can be anything.

It wasn't clear to me if immediately if the context would be the input element, or some other thing (i.e. Cassandra persistence metadata maybe?)

> That would require to have tuples as elements.

I was suggesting only providing a flow as a `FlowWithContext`.  If users wanted a normal flow they could drop down to it with `asSource`.  Maybe that's too much though.